Code/Framework/AzCore/AzCore/DOM/DomPath.cpp

```diff
@@ -135,6 +135,58 @@ namespace AZ::Dom
             },
             m_value);
     }
+
+    bool operator<(const AZ::Dom::PathEntry& lhs, const AZ::Dom::PathEntry& rhs)
+    {
+        if (lhs.IsIndex())
+        {
+            return lhs.GetIndex() < rhs;
+        }
+        else
+        {
+            return lhs.GetKey() < rhs;
+        }
+    }
+
+    bool operator<(const AZ::Dom::PathEntry& entry, size_t index)
+    {
+        return entry.IsIndex() && entry.GetIndex() < index;
+    }
+
+    bool operator<(size_t index, const AZ::Dom::PathEntry& entry)
+    {
+        return !entry.IsIndex() || index < entry.GetIndex();
+    }
+
+    bool operator<(const AZ::Dom::PathEntry& entry, const AZ::Name& key)
+    {
+        return entry < key.GetStringView();
+    }
+
+    bool operator<(const AZ::Name& key, const AZ::Dom::PathEntry& entry)
+    {
+        return key.GetStringView() < entry;
+    }
+
+    bool operator<(const AZ::Dom::PathEntry& entry, AZStd::string_view key)
+    {
+        return entry.IsIndex() ||
+            AZStd::lexicographical_compare(
+                entry.GetKey().GetStringView().begin(),
+                entry.GetKey().GetStringView().end(),
+                key.begin(),
+                key.end());
```

Comment:
This should use the string_view operator<
```suggestion
        return entry.IsKey() && entry.GetKey().GetStringView() < key;
```

Reply:
I didn't realize that strings were compared lexicographically. I thought there was some difference in something like `"ac" < "b" == false`. I see now that they use https://en.cppreference.com/w/cpp/algorithm/lexicographical_compare. Will get that swapped out.

Reply:
The [string_view](https://en.cppreference.com/w/cpp/string/basic_string_view/operator_cmp) and string C++ classe comparison operators are lexicographical.
What is isn't lexicographical is comparing of the raw `const char*` C type which is a numeric pointer check.